selectors/selectors.md

```diff
@@ -108,3 +108,271 @@ These components imply or expand into the following things:
 - Allow language-specific implementations of selectors (parsers, execution, etc).
 - (IMPORTANT) Well-designed set of test vectors representing a variety of use cases for IPLD Selectors.
 - A recommended structure for implementing a selector type, with an easy to use test suite.
+
+
+Selector types
+--------------
+
+### CID Selector
+
+Selects a single CID.
+
+
+### Path Selector
+
+Selects a single path segment.
+
+
+### Array Selector
+
+Selects a slice or single item of an array of items.
+
+
+### Recursive Selector
+
+Uses another Selector to select things recursively. It also takes a limit on how often this recursion should happen (the limit can be set to "infinity" to keep traversing until there are no matching paths left.
+
+The recursion may also be stopped by a given CID.
+
+
+### Selector Query
```

Comment:
is "query" the best name for this? could it just be "Selector List" or is there context I'm missing (very possible!)?

Reply:
Good catch, this is now the `CidRootedSelector`. That also explains why the examples below were wrong :)